src/WinRT.Runtime/Interop/EventSourceState{TDelegate}.cs

```diff
@@ -0,0 +1,162 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.ComponentModel;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
+#nullable enable
+
+namespace WinRT.Interop
```

Comment:
ABI?

Reply:
I thought you just wanted ABI for the implementation types specific to different event handler types? As in, whereas the base infrastructure would just be in WinRT.Interop. The rationale was that the former are tied to specific public types, whereas the latter are just part of the fundamental building blocks provided by CsWinRT to build stuff. I can change that though if you have a strong preference 🙂

Reply:
I am thinking anything that isn't a projected type that consumers use would go under the ABI namespace.  So the building blocks would be in the ABI namespace so users don't need to worry about seeing it or so.  The exception is if it is internal, then it doesn't really matter.

Reply:
Mmh wait, but we already have a bunch of stuff in WinRT.Interop that's just meant for advanced scenarios, like `IID`, I thought the namespace would just be for "advanced scenarios" in a way? And these types are also hidden. But I can move them if you really prefer to not have them here, just let me know what you think.

Reply:
I agree WinRT.Interop is like for advanced scenarios too.  IID was fine as that can be something some users may want to use.  And same goes for ActivationFactory.  But this I don't see what someone would want to use it for given it is all event implementation logic.  Thoughts?

Reply:
Sounds good to me! Done in 51e3e5740b3ca8a000d2510db3e48d9ef8301af7.